### ISSUE TITLE
Optional Url Params

### ISSUE THREAD

**Issue by luke-unifymonitor** [icd2k3/react-router-breadcrumbs-hoc#37]
Hi,

Love the project; thanks,

I have a problem with a path with an optional parameter.

When navigating to `/sensor/register` the breadcrumb shows:

> Register Sensor

but navigating to `/sensor/register/123` results in:

> Register Sensor / Register Sensor

I have the following path setup in my routes.config - the offending route is at the bottom...

```
const routesConfig = [
  {
    path: '/hubs',
    exact: true,
    component: MyHubsContainer,
    breadcrumb: 'My Hubs'
  },
  {
    path: '/hubs/register',
    exact: true,
    component: RegisterHubContainer,
    breadcrumb: 'Register Hub'
  },
  {
    path: '/hubs/:hubId',
    exact: true,
    component: HubDashboardContainer,
    breadcrumb: HubBreadcrumbItem
  },
  {
    path: '/hubs/:hubId/sensor/:sensorId/:sensorName',
    exact: true,
    component: SensorDetailContainer,
    breadcrumb: SensorBreadcrumbItem
  },
  {
    path: '/sensor/instructions',
    exact: true,
    component: InstructionsContainer
  },
  {
    path: '/sensor/register/:hubId?',
    exact: true,
    component: RegisterSensorContainer,
    breadcrumb: 'Register Sensor'
  }
]
```

```
const MainRoutes = (props) =>
  <Switch>
    {routesConfig.map((route, i) =>
      <Route
        exact={route.exact}
        key={route.path}
        path={route.path}
        component={route.component}
      />)}
  </Switch>
```

```
const Breadcrumbs = ({ breadcrumbs }) =>
  <div>
    {breadcrumbs.map((breadcrumb, index) => (
      <Link key={breadcrumb.key} to={breadcrumb.props.match.url || ''}>
        {breadcrumb} {console.log('breadcrumb', breadcrumbs.length, breadcrumb)}
      </Link>
    ))}
  </div>
```

Any ideas?

**Comment by icd2k3** [comment]
👋  hey @luke-unifymonitor!

This definitely seems like something we need to add support for.

I should have some time this weekend to take a look and hopefully come up with a plan. In the meantime, would maintaining 2 separate routes work?

Something like (untested):
```
const routesConfig = [
  //...
  {
    path: '/sensor/register',
    exact: true,
    component: RegisterSensorContainer,
    breadcrumb: 'Register Sensor',
  },
 {
    path: '/sensor/register/:hubId',
    exact: true,
    component: RegisterSensorContainer,
    breadcrumb: null,
  }
];
```

Not sure if that might work for the use case in the short term, but even if it does, it's obviously not ideal.

Sounds like I might need to come up with an additional param for `optionalParamBreadcumbs: <Array>` 🤔

**Comment by icd2k3** [comment]
Actually... this might be a cleaner solution. Instead of a `string` you could use a breadcrumb component that checks for the optional param.

Again this is untested, but I'll give it a shot later tonight to make sure it works:
```
const routesConfig = [
  {
    path: '/sensor/register/:hubId?',
    exact: true,
    component: RegisterSensorContainer,
    breadcrumb: RegisterSensorBreadcrumb,
  }
];

const RegisterSensorBreadcrumb = ({ match }) => {
  if (!match.params.hubId) { return null; }
  return (
    <span>Register Sensor</span>
  );
};
```

**Comment by icd2k3** [comment]
Gonna close this as the suggestion above should solve for this use case. If it doesn't, just let me know and I'll reopen this. Thanks!

**Comment by luke-unifymonitor** [comment]
Hi @icd2k3,

Apologies for not replying before.  We had a re-think for the application and no longer needed the paths that caused the problem.

However,  I did create this https://codesandbox.io/s/kok304yz1v and the console reads:

> Warning: Encountered two children with the same key, `/sensor/register/:hubId?`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.

Many thanks for your help.

**Comment by icd2k3** [comment]
Interesting... I'll look into this case in the near future. Thanks for the reply and the code sandbox!

**Comment by icd2k3** [comment]
Hey @luke-unifymonitor this key warning is fixed in `2.1.4`. Here is the updated code sandbox: https://codesandbox.io/s/13j7vr75yl
